juice/src/util.rs

```diff
@@ -72,6 +72,36 @@ pub fn cast_vec_usize_to_i32(input: Vec<usize>) -> Vec<i32> {
     out
 }
 
+// Formats tensor as a 2D matrix.
+pub fn format_tensor(tensor: &SharedTensor<f32>) -> String {
+    let native = native_backend();
+
+    let mut output = String::new();
+
+    if tensor.desc().len() == 1
+        || (tensor.desc().len() == 2 && (tensor.desc()[0] == 1 || tensor.desc()[1] == 1))
+    {
+        // One-dimensional, print as a single row.
+        let native_data: &[f32] = tensor.read(native.device()).unwrap().as_slice();
+        for v in native_data {
+            output += &format!("{:.5} ", v);
+        }
+        output += "\n";
+    } else {
+        // Use first dimension a row number.
+        let rows = tensor.desc()[0];
+        let columns = tensor.desc().iter().skip(1).fold(1, |acc, d| acc * d);
+        let native_data: &[f32] = tensor.read(native.device()).unwrap().as_slice();
+        for row in 0..rows {
+            for col in 0..columns {
+                output += &format!("{:.5} ", native_data[row * columns + col]);
+            }
+            output += "\n"; 
+        }
+    }
+    output
+}
```

Comment:
An impl of `std::fmt::Debug` might make sense

Reply:
There is already a Debug implementation for SharedTensor which currently only prints dimensions: https://github.com/spearow/juice/blob/2849c96c7eeb58e4049b7055f9db08302059c443/coaster/src/tensor.rs#L268. Do you want me to replace that?

Reply:
Not enterily sure tbh. Leave as is for now, it's too big of a change for this PR

Reply:
OK, I'll put a TODO.